core/services/gateway/handlers/handler.go

```diff
@@ -15,28 +15,33 @@ type UserCallbackPayload struct {
 	ErrMsg  string
 }
 
-// Handler implements service-specific logic for managing messages from users and nodes.
-// There is one Handler object created for each DON.
+// UserMessageHandler implements service-specific logic for managing messages from users and nodes.
+// There is one UserMessageHandler object created for each DON.
 //
-// The lifecycle of a Handler object is as follows:
+// The lifecycle of a UserMessageHandler object is as follows:
 //   - Start() call
 //   - a series of HandleUserMessage/HandleNodeMessage calls, executed in parallel
-//     (Handler needs to guarantee thread safety)
+//     (UserMessageHandler needs to guarantee thread safety)
 //   - Close() call
-type Handler interface {
+type UserMessageHandler interface {
```

Comment:
What are we actually gaining from this split? It seems that it only adds an extra cast and forces you to use a generic Service type in a map...